src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const obj = { ...state };
+
+  for (const element of actions) {
+    const { type, extraData, keysToRemove } = element;
+
+    switch (type) {
+      case 'addProperties': {
+        for (const key in extraData) {
```

Comment:
Consider using Object.assign instead of looping through every property for better performance